### PR TITLE
Better typing

### PR DESCRIPTION
**Pull request by fredj**
Small typing fixes found by setting `strict` to `true` in `tsconfig.json`

### CODE REVIEW

**Comment by fredj** [comment]
The strict mode yields around 3500 error but I hope most of them are easily fixable.

**Comment by fredj** [comment]
For example with this code:
```js
/**
 * The map pixel relative to the viewport corresponding to the original browser event.
 * @type {import("./pixel.js").Pixel}
 */
this.pixel_ = null;
```
typescript complains that _Type 'null' is not assignable to type 'number[]'.ts(2322)_
We could change the type to be `@type {null|import("./pixel.js").Pixel}`

**Comment by fredj** [comment]
I've added two commit that adds more typecast and mark properties as nullable

File: src/ol/PluggableMap.js [hunk 1]
```diff
@@ -604,7 +604,7 @@ class PluggableMap extends BaseObject {
     }
     const options = opt_options || /** @type {AtPixelOptions} */ ({});
```
Comment (tschaub) [inline src/ol/PluggableMap.js]: Not related to your change here, but I think we should also address type casts like the one above.  If an empty object is not assignable to `AtPixelOptions`, we shouldn’t be using it as one.

Comment (fredj) [reply src/ol/PluggableMap.js]: You're right: these typecast are not necessary. Typescript is able to deduce that an empty object is compatible with `AtPixelOptions`.

pull request to come.

Comment (fredj) [reply src/ol/PluggableMap.js]: See #10076

**Pull request merged**